walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -28,28 +28,45 @@
     public int compare(Event a, Event b) {
       return Long.compare(a.getWhen().start(), b.getWhen().start());
     }};
-
-  public Collection<TimeRange> query(Collection<Event> eventCollection, MeetingRequest request) { 
+    
+  public Collection<TimeRange> query(Collection<Event> eventCollection, MeetingRequest request) {
+    Collection<String> attendees = new ArrayList<String>(); 
+    attendees.addAll(request.getOptionalAttendees());
+    attendees.addAll(request.getAttendees()); 
+    if (request.getOptionalAttendees().isEmpty()) {
+      attendees.removeAll(request.getOptionalAttendees());
+    }
+    if (request.getAttendees().isEmpty()) {
+      attendees.removeAll(request.getAttendees()); 
+    }
+    if (queryHelper(eventCollection, request, attendees).isEmpty()) {
+      attendees.removeAll(request.getOptionalAttendees());
+      return queryHelper(eventCollection, request, attendees);
+    }
+    return queryHelper(eventCollection, request, attendees);   
+  }
+  
+  private static Collection<TimeRange> queryHelper(Collection<Event> groupOfEvents, MeetingRequest request, Collection<String> attendees) {
```

Comment:
I made this note below, but I'll echo it here.

`Collection` is a really weird argument type to accept. It makes no guarantees about the performance of operations on it, which is not generally what you want when you know you'll iterate through them.

Consider accepting List<> or ArrayList<>.